### PR TITLE
[RLlib] APPO/PPO Atari release tests: switch over to new API stack.

### PR DESCRIPTION
**Pull request by sven1977**
APPO/PPO Atari release tests: switch over to new API stack.

Results:
**APPO on Pong-v5:**
image
Above: Pong-v5 (no frameskip, deterministic) on APPO: red: tf2 + tracing; blue: torch; x-axis is relative time (in hours)

Config used:
```
# This can reach 18.0 reward in ~10 minutes on 4x M60 GPUs
# with 30 rollout workers, 4 learning workers, and 8 envs per rollout worker.
pong-appo:
    env: ALE/Pong-v5
    run: APPO
    stop:
        episode_reward_mean: 18.0
        timesteps_total: 20000000
    config:
        # Works for both torch and tf.
        framework: torch
        # Make analogous to old v4 + NoFrameskip.
        env_config:
            frameskip: 1  # no frameskip
            full_action_space: false
            repeat_action_probability: 0.0  # deterministic
        vtrace: true
        use_kl_loss: false
        rollout_fragment_length: 50
        train_batch_size: 4000
        lr: 0.0006
        # On a 32 CPU machine (g3.2xlarge), we use 30 CPUs for the rollout workers
        # and 2 for the learner workers.
        num_workers: 31
        broadcast_interval: 1
        max_sample_requests_in_flight_per_worker: 1
        num_envs_per_worker: 8
        num_sgd_iter: 2
        vf_loss_coeff: 1.0
        clip_param: 0.3
        num_gpus: 0

        model:
            dim: 42
            conv_filters: [[16, 4, 2], [32, 4, 2], [64, 4, 2], [128, 4, 2]]
            conv_activation: relu

        # Run with Learner API.
        _enable_learner_api: true
        grad_clip: 10.0
        # Use N Learner worker on the GPU
        num_learner_workers: 4
        num_gpus_per_learner_worker: 1
        # Since we are using learner workers, the driver process does not need
        # a CPU in particular.
        num_cpus_for_local_worker: 1

        # Run with RLModule API.
        _enable_rl_module_api: true
        # Need to unset this b/c we are using the RLModule API, which
        # provides exploration control via the RLModule's `forward_exploration` method.
        exploration_config: {}
```

**PPO on Breakout-v5**
image
Above: Breakout-v5 (no frameskip, deterministic) on PPO: orange: torch + new API stack + 4 GPUs; blue: torch + old API stack + 1GPU; x-axis is relative time (in hours)

Config used:
```
ppo-breakoutnoframeskip-v5:
    env: ALE/Breakout-v5
    run: PPO
    stop:
        sampler_results/episode_reward_mean: 100.0
        timesteps_total: 10000000
    config:
        # Make analogous to old v4 + NoFrameskip.
        env_config:
            frameskip: 1
            full_action_space: false
            repeat_action_probability: 0.0
        lambda: 0.95
        kl_coeff: 0.5
        clip_rewards: true
        clip_param: 0.1
        vf_clip_param: 10.0
        entropy_coeff: 0.01
        train_batch_size: 5000
        rollout_fragment_length: auto
        sgd_minibatch_size: 500
        num_sgd_iter: 10
        num_workers: 59
        num_envs_per_worker: 1
        model:
            vf_share_layers: true
            conv_filters: [[16, 4, 2], [32, 4, 2], [64, 4, 2], [128, 4, 2]]
            conv_activation: relu

        min_time_s_per_iteration: 30
        lr: 0.0001

        grad_clip: 100.0

        # New stack (comment these all out when running on old stack):
        _enable_learner_api: true
        _enable_rl_module_api: true
        num_learner_workers: 4
        num_gpus_per_learner_worker: 1
        num_cpus_for_local_worker: 1
        exploration_config: {}

        # Old stack (set to 0 when running on new stack):
        num_gpus: 1

```

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by klekkala** [comment]
Which version of ray are you using for this? I built ray from appo_ppo_new_stack_release_test branch in @sven1977 's git repo. but the results are nowhere close to what you are getting.

**Comment by sven1977** [comment]
Hmm, strange. Just ran these yesterday:
The above plots were run on the branch: `sven1977/new_stack_appo_atari_benchmark_04`, which I merged yesterday into master.
Other than the rllib folder, which was linked against this branch, everything else was still on a Docker image from a few weeks ago.

I'll have to run more benchmarking experiments anyways and will keep you posted.
This branch here will replace our nightly release tests on APPO/PPO using the new stack, so it will make sure, everything is sound.

**Comment by klekkala** [comment]
Ok. thanks for your prompt response. I really need this new stack API for my research and need to run some experiments which I plan to submit to CoRL. Not sure what is wrong with my replication. I

File: release/release_tests.yaml [hunk 1]
```diff
@@ -3581,7 +3581,7 @@
         cluster_env: app_config.yaml
         cluster_compute: 1gpu_24cpus_gce.yaml
 
-- name: rllib_learning_tests_appo_tf
+- name: rllib_learning_tests_appo_tf2
```
[the remark this answers is not in the document]
Comment (sven1977) [reply release/release_tests.yaml:3584]: Ok, thanks for raising this. I can just leave it. It's just a name.

[the remark this answers is not in the document]
Comment (sven1977) [reply release/release_tests.yaml]: fixed

File: release/release_tests.yaml [hunk 2]
```diff
@@ -4058,7 +4171,8 @@
         cluster_env: app_config.yaml
         cluster_compute: 2gpus_32cpus_gce.yaml
 
-- name: rllib_learning_tests_ppo_torch
+# TODO (sven): Remove this test once we are on new stack by default for APPO.
+- name: rllib_learning_tests_ppo_old_stack_torch
```
Comment (avnishn) [inline release/release_tests.yaml:4175]: so long as we are ok with breaking preset this should be fine.

**Review by avnishn — APPROVED** [review]

**Pull request merged**